### PR TITLE
feat: Enable backstage to connect to local cluster

### PR DESCRIPTION
**Pull request by tumido**
Signed-off-by: Tomas Coufal <tcoufal@redhat.com>

## Description of the change

If user has the kubernetes plugin installed in Backstage, this allows him to easily connect it to the local cluster. 

## Existing or Associated Issue(s)

n/a

## Additional Information

n/a

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [helm-docs](https://github.com/norwoodj/helm-docs) utility can be used to generate the necessary content. Use `helm-docs --dry-run` to preview the content.
- [x] JSON Schema generated.
- [x] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

**Comment by tumido** [comment]
@sabre1041 PTAL

File: charts/backstage/values.yaml [hunk 1]
```diff
@@ -99,6 +99,10 @@ serviceAccount:
 
 securityContext: {}
 
+## Configure backstage to connect to your local cluster
+kubernetesPlugin:
```
Comment (gorkem) [inline charts/backstage/values.yaml:103]: This only enables local cluster connection should we rename this to reflect that it is in fact only the cluster it is deployed to ?

**Comment by gorkem** [comment]
other than one naming suggestion should be good to go.

**Comment by tumido** [comment]
@gorkem what naming suggestion did you want to suggest? I see no comments in here... :slightly_frowning_face:

**Comment by gorkem** [comment]
@tumido funny somehow I managed to confuse myself. Submitted the review now.

File: charts/backstage/values.yaml [hunk 2]
```diff
@@ -99,6 +99,10 @@ serviceAccount:
...
+  enabled: false
+
```
[the remark this answers is not in the document]
Comment (tumido) [reply charts/backstage/values.yaml:105]: I see... What would you suggest we name it?

What about something like this:

```
kubernetesPlugin:
    connectToLocalCluster: true
```

I don't know, it's too verbose for me, but I really don't have any idea how to formulate it better.

[the remark this answers is not in the document]
Comment (gorkem) [reply charts/backstage/values.yaml:105]: Perhaps just `connectToLocal`.

**Comment by tumido** [comment]
@gorkem updated adopting all your suggestions, please review again. :+1:

**Comment by raffaelespazzoli** [comment]
As it was discussed in another thread, I believe we should fully externalize the backstage configuration instead of emerging it piecemeal as helm parameters. So in general I am contrary to the approach suggested here. 
In the specific, this configuration is openshift specific, when it could just be kubernetes generic and does not take into account for multi-cluster deployments.
I don't know who is governing this project at this point but if it was me, I'd reject this contribution. @tumido this is nothing personal  you are doing a good job. But this direction just does not fit the model I'd use.

**Comment by tumido** [comment]
Hey @raffaelespazzoli can you please point me to that another thread you mentioned?

By externalizing the configuration you refer to `app-config.yaml` or the helm chart as a whole?

Yes, this doesn't take into account multicluster deployment since we have no notion of external clusters. However so far our helm chart didn't allow any kubernetes configuration, so I don't see an issue with starting with the local cluster. Also `app-config.yaml` changes introduced in this PR allow merging with another potential externalized config file in case we want to go that route to provide a baseline + allow further customization.

The idea of this PR is to allow backstage to connect to the local cluster, therefore I think its helpful to use Helm to create appropriate `ClusterRole` and `ClusterRoleBinding` in the cluster. You can't do that from within an externalized Backstage config file.

**Review by gorkem — APPROVED** [review]

**Comment by sabre1041** [comment]
@tumido should we close this PR?

**Comment by tumido** [comment]
Yup, closing

**Pull request closed without merging**